### DISCUSSION TITLE
What are good values to use for a set of ~15,000 images?

### DISCUSSION THREAD

**Discussion by zylervega** [TheLastBen/fast-stable-diffusion#965] [Q&A]
I have ~15,000 images.  

What would you suggest I use for UNet_Training_Steps and Text_Encoder_Training_Steps?

**Comment by TheLastBen** [comment]
15k images is a bit much, do this : 
- Cherry pick 40 images, rename them into a nonexisting word, upload them as instance images.
-  The rest of the images upload them as concept images.
- Set UNet training to 4000 steps
- Text_encoder to 650 steps
- Concept_text_encoder to 1000 steps

For what I understand, you want to train a style and not specific faces, so tick the box Style_training.

Test the model, and report about the result.

**Reply by zylervega** [reply]
> 15k images is a bit much, do this :
> 
> * Cherry pick 40 images, rename them into a nonexisting word, upload them as instance images.
> * The rest of the images upload them as concept images.
> * Set UNet training to 4000 steps
> * Text_encoder to 650 steps
> * Concept_text_encoder to 1000 steps
> 
> For what I understand, you want to train a style and not specific faces, so tick the box Style_training.
> 
> Test the model, and report about the result.

Hm.  I can't cherry pick 40 for this project.

What I have is a screensaver I have been building over the last 20 years.  In it I have ~15k unique images.  What I wish is to train a model on these images and then produce output based on those images.  They span many different styles and media (photo, illustration, etc).  I do not care if the output looks recognizable as I'd be using it to composite abstract art.

I wonder if there is some other training method that would make sense?
Or, I wonder if your response was predicated on the assumption that "good" output is an important factor?

I will try the style checkbox and generally I will report results here for those interested and additional feedback, thanks!

**Reply by TheLastBen** [reply]
as I said, cherry pick 40, then upload the rest as "concept_images", the model will still get trained on all 15.000 pictures.

**Reply by zylervega** [reply]
> * The rest of the images upload them as concept images.

How/where does one upload concept images?  I will try this thanks

**Reply by TheLastBen** [reply]
I added a cell right below the cell "instance images", get the update using the official link : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast-DreamBooth.ipynb

**Reply by nawnie** [reply]
> I added a cell right below the cell "instance images", get the update using the official link : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast-DreamBooth.ipynb

you do this to me on purpose don't you. every time i get things nailed down LOL

**Comment by nawnie** [comment]
So with that many images, sorry Ben, id suggest looking at somethibg like every dream, do note these bigger fine tunners require 24 gigs to run. 

Now to force Ben's to work "with limitations" check out my Futurama post and read the images,

Text encoder needs to be REAL low, like 2 scans of every picture low. 

You will need to add class images to an extent, just throw them in your instance images folder, for 15k use 300

The amount of subjects and styles you are trying to attempt will change this, at the end of the day it's doable but your looking for a needle in a hay stack, once images are similar you will need to slowly progress at 15000 steps at a time. If things bleed start over, and do less text encoder.

Nows a good time to mention colab will likley kick you around 50k steps depending on how busy they are. 

Not saying either is better but you can only push so much outta the vram that Ben is shooting for.

If this must be done one Ben's, in the train unet change batch size to 2 and learning rate to 1e-6. Set your check points to start at 30k and do one every 15k after wards,

Set your total steps to infinite and text encoder to 300 steps, best to under train now and get your 30k in. 

Not going to lie its probably going to take weeks of tweaking

-------------------------
Welp nvm i have no idea now lol

**Reply by zylervega** [reply]
Thank you, I will also try your suggestion.  And I am looking into borrow/renting some time on an a100 or something to try EveryDream.

> Text encoder needs to be REAL low, like 2 scans of every picture low.

So, the value to 2, or to 2 * [total number of images]?  I imagine it's the 2nd one but wanted to check.

> You will need to add class images to an extent, just throw them in your instance images folder, for 15k use 300

These are images separate from my dataset?  or are these images characteristic of my data set / the "hand-picked" that Ben mentioned?

> you will need to slowly progress at 15000 steps at a time

15,000 steps is a lot more steps than I tried on the first round; good to know.

> If this must be done one Ben's, in the train unet change batch size to 2 and learning rate to 1e-6. Set your check points to start at 30k and do one every 15k after wards,

Got it.

> Set your total steps to infinite and text encoder to 300 steps, best to under train now and get your 30k in.

This seems to be different than the amount you suggested above for text encoder; am I misunderstanding?

-

And one more q- for the method you are proposing, do all 15k images go into Instance Images / ignore Concept Images?

Thanks!

**Comment by nawnie** [comment]
Honestly i have not played with the concept folder, it's new, as far as the
text encoder goes I had forgotten it was 15k images that's way too many
steps, I'd forgo training the text encoder itself and maybe throw half your
images into the concept folder and train that encoder as it seems it would
use more broad terms.

Since we are not training the text encoder make sure you name your files,
in this case id put them all in separate folders grouping like ones then
label them tag1_tag2_tag3.png.

Bulk rename utility on windows or bulk file renamer on gdrive are your
friend here

**Reply by zylervega** [reply]
> Since we are not training the text encoder make sure you name your files,
in this case id put them all in separate folders grouping like ones then
label them tag1_tag2_tag3.png.

I can't manually name/tag 15,000 images, that's why I was hoping to use an automated process utilizing AI to tag them

**Comment by nawnie** [comment]
https://github.com/victorchall/EveryDream

Use autoCaption ipynb. May need to change q factor and length sadly it's
not a tagger but a full captioner.

However if you can separate them into diffrent sub categories bulk rename
utility works very well too.

If you find another good autocaption for bulk files please share.

**Reply by artificialguybr** [reply]
> https://github.com/victorchall/EveryDream Use autoCaption ipynb. May need to change q factor and length sadly it's not a tagger but a full captioner. However if you can separate them into diffrent sub categories bulk rename utility works very well too. If you find another good autocaption for bulk files please share.
> […](#)
> On Wed, Dec 14, 2022, 2:05 PM zylervega ***@***.***> wrote: Since we are not training the text encoder make sure you name your files, in this case id put them all in separate folders grouping like ones then label them tag1_tag2_tag3.png. I can't manually name/tag 15,000 images, that's why I was hoping to use an automated process utilizing AI to tag them — Reply to this email directly, view it on GitHub <[#965 (reply in thread)](https://github.com/TheLastBen/fast-stable-diffusion/discussions/965#discussioncomment-4403588)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AZPYLSB22LYLNP73BSOJWUTWNISAXANCNFSM6AAAAAAS2WPVSE> . You are receiving this because you commented.Message ID: <TheLastBen/fast-stable-diffusion/repo-discussions/965/comments/4403588@ github.com>

For Blip 

https://colab.research.google.com/drive/1NZ4bA-lhi1vsS_-VDSCzR_WMOGqRWJgr

**Reply by nawnie** [reply]
Not sure if you checked the repo I linked but it's the same code, one I linked has a few more data set tools as well though. Looking for something non blip or a list of custom made blip models personally

**Reply by artificialguybr** [reply]
https://colab.research.google.com/github/pharmapsychotic/clip-interrogator/blob/main/clip_interrogator.ipynb#scrollTo=rbDEMDGJrJEo

> Not sure if you checked the repo I linked but it's the same code, one I linked has a few more data set tools as well though. Looking for something non blip or a list of custom made blip models personally

maybe clip interrogator that use the latest clip released by stability?
https://colab.research.google.com/github/pharmapsychotic/clip-interrogator/blob/main/clip_interrogator.ipynb#scrollTo=rbDEMDGJrJEo
